### ISSUE TITLE
Upgrade from 0.23.0 to 0.24.0 "found backend type fs, expected xl or xl-single"

### ISSUE THREAD

**Issue by graelo** [wandb/local#98]
Hi, I'm using Docker and trying to upgrade from `0.23.0` to `0.24.0`, but minio does not start and encourages me to migrate its configuration.

```text
*** Access W&B at http://localhost:8080
*** Still waiting...
mc: <ERROR> Unable to get service status. Get "http://127.0.0.1:9000/minio/admin/v3/info": dial tcp 127.0.0.1:9000: connect: connection refused.
!!! Something's not right
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
```

Regarding the warning, indeed, `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY` are still used in `/vol/env/minio_creds.env` and `/vol/env/overrides.env`. I guess this is related to the recent change in minio (I use it for other stuff), but I could not see a mention of this upgrade in this repo.

Is this minio config problem something you are aware of when migrating from `0.23.0` to `0.24.0`?

FYI, I created my database wth `0.19.0` a few months ago, and migrated it to `0.20.0`, then `0.22.0`, then `0.23.0` without issues, so kudos for the migration work so far!

Thanks a lot!

PS: By the way, I've made sure that user `wandb` in the container owns `/vol`.

**Comment by MBakirWB** [comment]
Hi @graelo , thank you for writing in and reporting this. We will review and get back to you this week with an update.

**Comment by graelo** [comment]
Thanks @MBakirWB!

**Comment by yogeshg** [comment]
> Is this minio config problem something you are aware of when migrating from 0.23.0 to 0.24.0?

I was able to use 0.24.1 in my machine, and while minio complains about this, it's not an error, but just a warning. After execing into docker image, I ran  `cat /var/log/minio.log` and saw the following:
```
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
Warning: Default parity set to 0. This can lead to data loss.
MinIO Object Storage Server
Copyright: 2015-2022 MinIO, Inc.
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Version: RELEASE.2022-11-17T23-20-09Z (go1.19.3 linux/amd64)

Status:         1 Online, 0 Offline. 
API: http://172.17.0.2:9000/  http://127.0.0.1:9000/ 
SQS ARNs: arn:minio:sqs:wandb-local:_:redis 
Console: http://172.17.0.2:35965/ http://127.0.0.1:35965/ 

Documentation: https://min.io/docs/minio/linux/index.html

 You are running an older version of MinIO released 1 week ago 
 Update: Run `mc admin update`
```

I suspect the following might be more relevant:
```
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
```

As next steps, could you please:
1. update to `0.24.1` (un-related, but good to be on same page)
2. provide the minio log, by doing `cat /var/log/minio.log` in the running container?
3. provide all env variables, by doing `env|sort` In the running container?
4. provide the screenshot of wandb volume data by going to `Docker` UI in your machine > `Volumes` > `wandb` > `Data` > `minio` and then expanding the second level folders in minio. e.g:
<img width="645" alt="Screen Shot 2022-11-28 at 2 56 56 PM" src="https://user-images.githubusercontent.com/1059168/204398286-485594f6-1de9-4abe-91dc-80b414c5fe44.png">

**Comment by yogeshg** [comment]
Could you please also include a screenshot where a few wandb_artifacts are fully open?
https://github.com/wandb/local/issues/98

I'm wondering if you're you using `filesystem` mode on your minio...

**Comment by graelo** [comment]
Hi, thanks a lot @yogeshg 

I upgraded to `0.24.1`, here is my `/var/log/minio.log`:

<details>

```text
$ cat /var/log/minio.log
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
WARNING: MINIO_ACCESS_KEY and MINIO_SECRET_KEY are deprecated.
         Please use MINIO_ROOT_USER and MINIO_ROOT_PASSWORD
ERROR Unable to use the drive /vol/minio: Drive /vol/minio: found backend type fs, expected xl or xl-single - to migrate to a supported backend visit https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html: Invalid arguments specified
```

</details>

Here is my `env | sort`:

<details>

```text
env | sort
AWS_REGION=wandb-local
CONDA_MD5=8c69f65a4ae27fb41df0fe552b4a8a3b
CONDA_VERSION=4.10.3
container=podman
DEBIAN_FRONTEND=teletype
GORILLA_ANALYTICS_SINK=mysql://wandb_local:wandb_local@127.0.0.1:3306/wandb_local
GORILLA_AUTH_JWK_URL=http://127.0.0.1:8083/api/jwks.json
GORILLA_AUTH_METHOD=
GORILLA_CASBIN_ADDRESS=mysql://wandb_local:wandb_local@127.0.0.1:3306/wandb_local
GORILLA_DEFAULT_REGION=minio-local
GORILLA_EMAIL_SINK=https://api.wandb.ai/email/dispatch
GORILLA_FILE_HOST=http://localhost:8080
GORILLA_FILE_METADATA_SOURCE_INTERNAL=redis://127.0.0.1:6379/filemetadata
GORILLA_FILE_METADATA_SOURCE=redis://127.0.0.1:6379/filemetadata
GORILLA_FILE_STORE_IS_PROXIED=true
GORILLA_FILE_STORE=s3://8U4bhI7eDvvIILsJQ9Q6xQ==:0ZpzGR5UNEcn9OWHTW5Tiy56UOVbrCenanS3rWmlqTM=@127.0.0.1:9000/local-files
GORILLA_FILE_STREAM_STORE_ADDRESS=mysql://wandb_local:wandb_local@127.0.0.1:3306/wandb_local
GORILLA_FRONTEND_HOST=http://localhost:8080
GORILLA_GCP_PROJECT=dev~wandb-local
GORILLA_HISTORY_STORE=mysql://wandb_local:wandb_local@127.0.0.1:3306/wandb_local
GORILLA_LICENSE_CERT_PATH=/var/app/jwks.json
GORILLA_LICENSE=eyJhbGciOiJS...[snip]...qdiyASPIN-QTvbteY9mLJs__Gm4ngHPVUTjc2okr2cT3eD1J8KuVBosdz94q5IqdHCCDAxa3IgImLAzxno1VxiJr_3lw
GORILLA_LIMITER=redis://127.0.0.1:6379
GORILLA_LOCKER=redis://127.0.0.1:6379
GORILLA_MAX_CLI_VERSION=0.13.3
GORILLA_METADATA_STORE=mysql://wandb_local:wandb_local@127.0.0.1:3306/wandb_local
GORILLA_MIN_CLI_VERSION=0.8.20
GORILLA_OIDC_CLIENT_ID=
GORILLA_OIDC_ISSUER=
GORILLA_OIDC_SECRET=
GORILLA_ONPREM_API_KEY_PREFIX=local
GORILLA_ONPREM=true
GORILLA_PORT=8081
GORILLA_SCHEMA_FILE=/schema.graphql
GORILLA_SENTRY_DSN=
GORILLA_SENTRY_ENVIRONMENT=onprem-local
GORILLA_SESSION_KEY=QhVJsQxeStZEUCczSngiWM7kTIfxUQZt0XxvURMDdYs=
GORILLA_SESSION_LENGTH=720h
GORILLA_SETTINGS_CACHE=redis://127.0.0.1:6379
GORILLA_SLATE_PATH=/usr/local/bin/slate-linux
GORILLA_STATIC_HOST=http://localhost:8080
GORILLA_STATSD_PORT=0
GORILLA_STORAGE_BUCKET=s3://local-files
GORILLA_SWEEP_PROVIDER=http://127.0.0.1:8082
GORILLA_TASK_QUEUE=noop://
GORILLA_TRACER=noop://
HOME=/home/wandb
HOST=http://localhost:8080
HOSTNAME=6c99fc844546
INITRD=no
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_ALL=en_US.UTF-8
LC_CTYPE=en_US.UTF-8
LOCAL_BUILD_INFO_PATH=/home/wandb/build.yaml
LOCAL_CONTAINER_ENVIRONMENT_PATH=/etc/container_environment
LOCAL_ENV_PATH=/vol/env
LOCAL_LICENSE_CERT_PATH=/var/app/jwks.json
LOCAL_RESTART_CMD=restart.sh
LOCAL_SECURE=false
LOCAL_UI_PATH=/var/app/local
LOCAL_USER_SETTINGS_ERROR_CACHE_PATH=/home/wandb/user_settings_error_cache.json
MINIO_ACCESS_KEY=8U4bhI7eDvvIILsJQ9Q6xQ==
MINIO_CLIENT_RELEASE=2022-11-17T21-20-39Z
MINIO_CLIENT_SHA=6cf8c395d7a2aa90a27690eb9174124da49a438b
MINIO_HTTP_TRACE=/var/log/minio.log
MINIO_REGION=wandb-local
MINIO_SECRET_KEY=0ZpzGR5UNEcn9OWHTW5Tiy56UOVbrCenanS3rWmlqTM=
MINIO_SERVER_RELEASE=2022-11-17T23-20-09Z
MINIO_SERVER_SHA=d8b540f95108f8304b1f2b7c89f9175dac95324c
MYSQL_DATABASE=wandb_local
MYSQL_PASSWORD=wandb_local
MYSQL_USER=wandb_local
PATH=/opt/conda/bin/:/opt/conda/envs/base/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PROJECT_ID=playground-111
PWD=/home/wandb
PYTHONDONTWRITEBYTECODE=true
PYTHON_VERSION=39
SHLVL=0
TERM=xterm
_=/usr/bin/env
```

</details>

Here is the content of my `minio`:
(you will notice I don't have a `pool.bin` folder)

<details>

```text
$ tree -L 2 -a minio
minio
├── local-files
│   ├── myuser
│   ├── wandb-analytics-events
│   └── wandb_artifacts
└── .minio.sys
    ├── buckets
    ├── config
    ├── format.json
    ├── ilm
    ├── multipart
    ├── tmp
    └── tmp-old

11 directories, 1 file
```

</details>

Here is the content of `minio/.minio.sys/buckets`, I provide because I'm noticed that it contains nested `.minio.sys/buckets`:

<details>

```text
$ tree -L 2 -a minio/.minio.sys/buckets
minio/.minio.sys/buckets
├── .background-heal.json
├── .bloomcycle.bin
├── local-files
│   ├── bosr
│   ├── .metadata.bin
│   ├── .usage-cache.bin
│   ├── wandb-analytics-events
│   └── wandb_artifacts
├── .minio.sys
│   └── buckets
├── .tracker.bin
├── .usage-cache.bin
└── .usage.json

6 directories, 7 files
```

</details>

Just FYI, I started my database with 0.19.0, or with a slightly earlier version, and I did not modify my wandb minio config.

**Comment by yogeshg** [comment]
@graelo could you please try release 0.25.0 ?
https://github.com/wandb/local/releases/tag/0.25.0
If that doesn't work for you, I'll need to look at the full contents of the bucket, perhaps using the `find` command, if privacy is an issue, you could send it to support@ for wandb, and mention the issue number in subject, sth like "Support for https://github.com/wandb/local/issues/98".
I was able to reproduce the issue by using wandb/local version 0.11.0, creating some runs and artifacts and then using 0.24.0 or 0.24.1. I've fixed that in version 0.25.0. Assuming the fix works for you, this is happening because of [a breaking change in minio](https://github.com/minio/minio/pull/15929).

**Comment by graelo** [comment]
@yogeshg and yes it worked flawlessly!

**Comment by MBakirWB** [comment]
Hi @graelo, we are happy to hear that the released fix resolved your issue. Thanks again for flagging this for our team to review. Please do reach out again anytime we could be of help.

**Comment by graelo** [comment]
I guess I can now close the issue.

**Comment by yogeshg** [comment]
This affects users who've first adopted historical versions, like 0.11.0 and upgraded to [0.24.0](https://github.com/wandb/local/releases/tag/0.24.0) and [0.24.1](https://hub.docker.com/layers/wandb/local/0.24.1/images/sha256-75f4888fd9d96aa38e849580672a824fac4db1a10e35a4e996c54944178e723d?context=explore). It has been fixed in [0.25.0](https://github.com/wandb/local/releases/tag/0.25.0).
